libraries/AP_HAL_ChibiOS/hwdef/AnyleafH7/README.md

```diff
@@ -0,0 +1,75 @@
+# AnyLeaf Mercury H7 Flight Controller
+
+The Mercury H7 is a flight controller produced by [AnyLeaf](http://www.anyleaf.org/).
+
+## Features
+    Processor
+        STM32H743 32-bit processor
+    Sensors
+        ICM42688 Acc/Gyro with dedicated 32.768kHz crystal
+        DPS310 barometer
+    Power
+        2S - 6S Lipo input voltage with voltage monitoring
+        9V, 3A supply for powering video transmitters
+        5V, 2A supply for powering servos and electronics
+        3.3V, 1A supply for powering electronics
+    Interfaces
+        8x Bidirectional-DSHOT, or PWM-capable motor outputs
+        1x CAN-FD port for external peripherals
+        1x DJI-format Vtx connector
+        5x UARTs/serial for external peripherals
+        1x I2C for external peripherals
+        USB-C port
+        All UARTS support hardware inversion. SBUS, SmartPort, and other inverted protocols work on any UART without “uninvert hack”
+        Onboard ExpressLRS radio tranceiver
+
+## Pinout
+
+Pins and connector values are labeled on the flight controller PCB.
+
+## UART Mapping
+
+The UARTs are marked Rn and Tn in the above pinouts. The Rn pin is the
+receive pin for UARTn. The Tn pin is the transmit pin for UARTn.
+
+ - SERIAL0 -> USB
```

Comment:
needs dma capability addeds for each serial port

Reply:
How do I do that? Isn't that handled by AP? (They are all obviously DMA capable)

Reply:
first you have to include the bootloader files in this PR....it wont build without them....then you  run a waf configure and look at the hw.dat file produced to see what DMA channels get assigned ...also needed to assure that bidir requests are implemented for outputs in the configure....shared channels in bi-dir is a no-no

Reply:
Are you referring to `hwdef-bl.dat`? That is in the PR. (If it's something else, LMK!)

I'm still not quite sure how to follow those instructions, but I can tell you that these are all on General Purpose STM32 timers; that motors 1-4 are on the same timer, and motors 5/6 and 7/8 are each on another timer. They are all capable of Burst DMA, and bidirectional DSHOT.

Reply:
Where does this place the `hw.dat` Unable to find after running `./waf configure AnyleafH7` and using `grep`.

Reply:
![Screenshot from 2024-01-18 10-28-59](https://github.com/ArduPilot/ardupilot/assets/6076534/02e1f053-5dd3-4c87-b772-85a627fe3b8a)

Reply:
nvm; found it! `/build/board name`

Reply:
Here's the part relevant to UART and Motors:

```c
SERIAL_ORDER OTG1 USART1 USART2 USART3 UART4 UART7 UART8
PB6 USART1_TX USART1
PB7 USART1_RX USART1
PA2 USART2_TX USART2
PA3 USART2_RX USART2
PD8 USART3_TX USART3
PD9 USART3_RX USART3
PC10 UART4_TX UART4
PC11 UART4_RX UART4
PB4 UART7_TX UART7
PB3 UART7_RX UART7
PE1 UART8_TX UART8
PE0 UART8_RX UART8
PD0 CAN1_RX CAN1
PD1 CAN1_TX CAN1
PC6  TIM3_CH1 TIM3  PWM(1)  GPIO(50)  BIDIR
PC7  TIM3_CH2 TIM3  PWM(2)  GPIO(51)  BIDIR
PC8  TIM3_CH3  TIM3  PWM(3)  GPIO(52)  BIDIR
PC9  TIM3_CH4  TIM3  PWM(4)  GPIO(53)  BIDIR
PE5  TIM15_CH1  TIM15  PWM(5)  GPIO(54)
PE6  TIM15_CH2  TIM15  PWM(6)  GPIO(55)
PE13 TIM1_CH3 TIM1 PWM(7) GPIO(56)
PE14 TIM1_CH4 TIM1 PWM(8) GPIO(57)
PE9 TIM1_CH1 TIM1 GPIO(58) ALARM
```

Reply:
sorry its really the hwdef.h file which is only visible after you have bootloaders, which you didnt at this point